src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -162,8 +182,16 @@ void IStreamsExecutor::Config::SetConfig(const std::string& key, const std::stri
         } else if (streams == ov::streams::AUTO) {
             // bare minimum of streams (that evenly divides available number of cores)
             _streams = GetDefaultNumStreams();
+            if (_threadBindingType == ThreadBindingType::HYBRID_AWARE) {
+                // In HYBRID_AWARE, nstreams are limited by nthreads, which means 1 stream has at least 1 thread.
+                _streams = LimitHybridStreams(_streams, _threads);
+            }
         } else if (streams.num >= 0) {
             _streams = streams.num;
+            if (_threadBindingType == ThreadBindingType::HYBRID_AWARE) {
+                // In HYBRID_AWARE, nstreams are limited by nthreads, which means 1 stream has at least 1 thread.
+                _streams = LimitHybridStreams(streams.num, _threads);
```

Comment:
Why this logic is applicable only for Hybrid-aware?

Reply:
Will move limitation logic to following PR and benefit all affinities.